From 4d842135ce677957aee8bd5bb47f875b0ece570a Mon Sep 17 00:00:00 2001
From: Stefano Karapetsas <stefano@karapetsas.com>
Date: Mon, 6 May 2013 14:56:58 +0200
Subject: [PATCH] Remove debug prints

---
 src/gpm-control.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/gpm-control.c b/src/gpm-control.c
index 9f70f7f..71f857c 100644
--- a/src/gpm-control.c
+++ b/src/gpm-control.c
@@ -220,7 +220,6 @@ gpm_control_suspend (GpmControl *control, GError **error)
 			&dbus_error );
     if (proxy == NULL) {
         egg_error("Error connecting to dbus - %s", dbus_error->message);
-        fprintf(stderr,"Error connecting to dbus - %s", dbus_error->message);
         g_error_free (dbus_error);
         return -1;
     }
@@ -323,7 +322,6 @@ gpm_control_hibernate (GpmControl *control, GError **error)
 
 #ifdef WITH_SYSTEMD_SLEEP
 	/* sleep via logind */
-	fprintf(stderr,"logind suspend\n");
 	proxy = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SYSTEM,
 			G_DBUS_PROXY_FLAGS_DO_NOT_LOAD_PROPERTIES,
 			NULL,
@@ -334,7 +332,6 @@ gpm_control_hibernate (GpmControl *control, GError **error)
 			&dbus_error );
     if (proxy == NULL) {
         egg_error("Error connecting to dbus - %s", dbus_error->message);
-        fprintf(stderr,"Error connecting to dbus - %s", dbus_error->message);
         g_error_free (dbus_error);
         return -1;
     }
-- 
1.8.2.1

