diff -burN a/plugins/media-keys/msd-media-keys-manager.c b/plugins/media-keys/msd-media-keys-manager.c
--- a/plugins/media-keys/msd-media-keys-manager.c	2012-02-28 23:45:55.000000000 +0100
+++ b/plugins/media-keys/msd-media-keys-manager.c	2012-08-03 22:51:11.815389305 +0200
@@ -862,6 +862,19 @@
         update_default_sink (manager);
 }
 
+static void
+on_control_stream_removed (GvcMixerControl     *control,
+                           guint                id,
+                           MsdMediaKeysManager *manager)
+{
+        if (manager->priv->stream != NULL) {
+                if (gvc_mixer_stream_get_id (manager->priv->stream) == id) {
+                        g_object_unref (manager->priv->stream);
+                        manager->priv->stream = NULL;
+                }
+        }
+}
+
 #endif /* HAVE_PULSE */
 
 static gint
@@ -1186,6 +1199,10 @@
                           "default-sink-changed",
                           G_CALLBACK (on_control_default_sink_changed),
                           manager);
+        g_signal_connect (manager->priv->volume,
+                          "stream-removed",
+                          G_CALLBACK (on_control_stream_removed),
+                          manager);
 
         gvc_mixer_control_open (manager->priv->volume);
 
